### ISSUE TITLE
[iOS] Wallet import : freezes the first time it's used (after giving access to the camera)

### ISSUE THREAD

**Issue by mvaivre** [alephium/mobile-wallet#262]
Screen stays blank and tapping the close button does nothing.
Let's see if this can be reproduced on the simulator.

**Comment by mvaivre** [comment]
UPDATE: Testing the single wallet branch. Now I get:

- I give permission to use the camera
- I see the QR code scanning screen briefly
- Then I'm redirected to the welcome screen 🤔

**Comment by nop33** [comment]
This is not reproducible on the simulator (I tried the import scan screen as well as the qr code scan screen from the dashboard button, in both cases I did not get redirected to the welcome screen).

Ideas of where to look: There was this issue where when the OS-permissions modal would pop up, the app would think that it went to the background, so the logic of resetting the state would be executed (in App.tsx). To fix this, I stored a value called `isCameraOpen` in the redux state and toggle it just before the camera is opened and back again when it's closing. I used this flag as a condition on the app state change logic (active/background). But it doesn't happen on the simulator so I cannot verify that the problem is here... :/
